src/main/daml/Daml/Finance/Holding/Util.daml

```diff
@@ -57,17 +58,27 @@ transferImpl this self Transferable.Transfer{actors; newOwnerAccount} = do
 
 -- | Default implementation of `acquire` from the Base interface.
 acquireImpl :
-  ( HasCreate t
+  ( HasSignatory t
+  , HasCreate t
   , HasField "lock" t (Optional Base.Lock)
   , HasFromInterface t Base.I, HasToInterface t Base.I
   ) => t -> Base.Acquire -> Update (ContractId Base.I)
 acquireImpl base Base.Acquire{newLockers; context; lockType} = do
+  let v = view $ toInterface @Base.I base
+  whenSome v.lock \lock -> do
+    lock.lockers === newLockers
+    lock.lockType === lockType
+    assertMsg "Contract must be locked." $ lock.lockType == Base.Reentrant
+    assertMsg "Contract must not be locked for this context." $ S.notMember context lock.context
   newLock <- case (base.lock, lockType) of
     (Some existingLock, Base.Reentrant) -> pure existingLock with
       context = context `S.insert` existingLock.context
     (None, _) -> pure Base.Lock with lockers = newLockers; context = S.singleton context; lockType
     _ -> abort "Contract is already locked."
-  toInterfaceContractId <$> create base with lock = Some newLock
+  let newBase = base with lock = Some newLock
+  newBaseCid <- toInterfaceContractId <$> create newBase
```

Comment:
You don't need this, you can replace the last line with `toInterfacecontractId <$> create newBase`